File: codes/hash_table/383.ransom-note/ransom-note.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> dict;
        for (const auto c : magazine) {
            dict[c]++;
        }

        for (const auto c : ransomNote) {
            auto iter = dict.find(c);
            if (iter == dict.end() || iter->second == 0) {
                return false;
            }
            iter->second--;
        }
        return true;
    }
};

File: codes/hash_table/383.ransom-note/ransom-note.go
package ransomnote
func canConstruct(ransomNote string, magazine string) bool {
	dict:=make(map[rune]int)
	for _,c:=range magazine{
		dict[c]++
	}

	for _,c:=range ransomNote{
		if dict[c]==0{
			return false
		}
		dict[c]--
	}
	return true
}